cmd/mattermost/commands/server.go

```diff
@@ -65,6 +67,19 @@ func runServer(configStore *config.Store, usedPlatform bool, interruptChan chan
 	// and also preserve a crash dump for later investigation.
 	debug.SetTraceback("crash")
 
+	defer func() {
```

Comment:
In a normal method exit this would get executed after `server.Shutdown` due to lifo nature of the defer queue.  I'm not clear on if/how that changes in a panic scenario.  If `server.Shutdown` does get called first then the logging engine will be shutdown before this log call.

Reply:
That's a great point. From what I can see it shuts down the advanced logging part, which I didn't actually test. I am a bit rusty on the logging part, but does it disable the normal logger too, when the advanced logging shuts down?

Reply:
It doesn't shut down the Zap logger, nothing does - it never exits cleanly but that doesn't matter so much since it writes synchronously.  However the Zap logger will be removed later this year so I think it's best to output the panic log before shutdown so it gets flushed.